stan/math/prim/functor/integrate_ode_rk45.hpp

```diff
@@ -120,26 +121,34 @@ std::vector<std::vector<return_type_t<T1, T2, T_t0, T_ts>>> integrate_ode_rk45(
 
   // avoid recording of the initial state which is included by the
   // conventions of odeint in the output
+  size_t timestep = 0;
   auto filtered_observer
       = [&](const std::vector<double>& coupled_state, double t) -> void {
     if (!observer_initial_recorded) {
       observer_initial_recorded = true;
       return;
     }
     observer(coupled_state, t);
+    timestep++;
```

Comment:
This looks like it's going beyond just changing error type.  Can you explain what these modifications are intended to do in the PR?

Reply:
This is so the error can be informative about where the ODE solve broke. It can print out 'failed to integrate to time 7.5' for instance, instead of saying 'failed to integrate hit max_num_steps'. It just seemed more informative.